src/Razor/Microsoft.AspNetCore.Razor.Tools/src/RewriteCssCommand.cs

```diff
@@ -57,28 +59,49 @@ protected override bool ValidateArguments()
 
         protected override Task<int> ExecuteCoreAsync()
         {
+            var allDiagnostics = new ConcurrentQueue<RazorDiagnostic>();
+
             Parallel.For(0, Sources.Values.Count, i =>
             {
                 var source = Sources.Values[i];
                 var output = Outputs.Values[i];
                 var cssScope = CssScopes.Values[i];
 
                 var inputText = File.ReadAllText(source);
```

Comment:
I'm surprised we've gotten away with this to be honest. Reading text directly from .NET APIs has so many issues when coupled with tooling.

One thing to consider (don't feel obligated to do this if you feel it isn't worth it) would be to use an actual Roslyn `SourceText` to contain the `inputText` and read the file from disk similarly to [how Roslyn does](https://github.com/dotnet/roslyn/blob/master/src/Workspaces/Core/Portable/Workspace/FileTextLoader.cs) (can't directly use the type because it's in workspaces but it's a lite type). That will ensure that large files/different encodings are properly understood and it would also provide you some nice APIs to avoid you needing the `GetParseItemSourceSpan` below at all (there's an API to get locations like that on SourceText).

Reply:
Great feedback, thanks. I've updated it to use `SourceText` and eliminated the manual line-number-computation code.

Reply:
Note that it's still using `FileStream` under the `SourceText`, but does so using identical open flags as the rzc "generate" command and hence shouldn't behave any differently in terms of locking etc.